### PR TITLE
The Big Refactor/Rework - Part 2: Reviews & Likes

### PR DESCRIPTION
**Pull request by maxkajiwara**
# Description
- Refactor the review modal with updated actions and props
- Update modal to close or trigger confirmation modal when user clicks outside of it
- Disable action buttons while waiting for actions to complete
- Fix some minor bugs
- Add like functionality to the review modal

Miscellaneous changes included in this PR:
- Scroll landing page to top when navigated to any other way than back/forward

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Change status
- [x] Complete, tested, ready to review and merge
- [ ] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

# How Has This Been Tested?
- [ ] Play around with adding, editing, and deleting reviews on the project page
- [ ] Trying clicking outside the review modal in various states (text field changed, nothing changed, deleting review, etc)
- [ ] Don't worry about the likes for now unless you feel like manually changing the review_id to test them

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] My code has been reviewed by at least one peer
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] There are no merge conflicts

### CODE REVIEW

File: reactapp/src/reducers/reviewReducer.js [hunk 1]
```diff
@@ -8,39 +8,35 @@ import {
...
 	DELETING_REVIEW,
 	DELETED_REVIEW,
 	DELETE_REVIEW_ERROR,
-	SHOW_REVIEW_MODAL
+	LIKING_REVIEW,
+	LIKED_REVIEW,
+	LIKE_REVIEW_ERROR
 } from '../actions';
 
 const initialState = {
```
Comment (michaelagard) [inline reactapp/src/reducers/reviewReducer.js]: instead of commenting out code, just delete it.

File: reactapp/src/actions/reviewActions.js [hunk 2]
```diff
@@ -12,20 +12,18 @@ export const GET_REVIEW_ID_ERROR = 'GET_REVIEW_ID_ERROR';
...
-// willDeleteReview
-export const WILL_DELETE_REVIEW = 'WILL_DELETE_REVIEW';
 // deleteReview
 export const DELETING_REVIEW = 'DELETING_REVIEW';
 export const DELETED_REVIEW = 'DELETED_REVIEW';
 export const DELETE_REVIEW_ERROR = 'DELETE_REVIEW_ERROR';
-// showReviewModal
-export const SHOW_REVIEW_MODAL = 'SHOW_REVIEW_MODAL';
+// likeReview
+export const LIKING_REVIEW = 'LIKING_REVIEW';
```
Comment (michaelagard) [inline reactapp/src/actions/reviewActions.js]: "Error: could not update like data" when trying to like my own review. Should we be able to attempt to review our own review?

File: reactapp/src/actions/reviewActions.js [hunk 3]
```diff
@@ -12,20 +12,18 @@ export const GET_REVIEW_ID_ERROR = 'GET_REVIEW_ID_ERROR';
...
+export const LIKED_REVIEW = 'LIKED_REVIEW';
+export const LIKE_REVIEW_ERROR = 'LIKE_REVIEW_ERROR';
```
Comment (michaelagard) [inline reactapp/src/actions/reviewActions.js]: image
When clicking off a modal, there shouldn't be any error message when I click open up the modal. Should clear error state.

**Review by fellanonymous — CHANGES_REQUESTED** [review]
added a few remarks.
please review

File: reactapp/src/actions/reviewActions.js (hunk 2 above)
Comment (maxkajiwara) [reply reactapp/src/actions/reviewActions.js]: Fixed!
Review modal had been set up to show own reviews as likable for testing purposes.

File: reactapp/src/actions/reviewActions.js (hunk 3 above)
Comment (maxkajiwara) [reply reactapp/src/actions/reviewActions.js]: Fixed!
Getting a review now clears the likingReviewError property in the redux store

**Review by fellanonymous — APPROVED** [review]

**Pull request merged**